src/linear_algebra/determinant.lean

```diff
@@ -344,4 +363,78 @@ begin
     exact hkx }
 end
 
+/-- The determinant of a 2x2 block matrix with the lower-left block equal to zero is the product of
+the determinants of the diagonal blocks. For the generalization to any number of blocks, see
+`matrix.upper_block_triangular_det`. -/
+lemma upper_two_block_triangular_det (A : matrix m m R) (B : matrix m n R) (D : matrix n n R) :
+  (matrix.from_blocks A B 0 D).det = A.det * D.det :=
+begin
+  unfold det,
+  rw sum_mul_sum,
+  let preserving_A : finset (perm (m ⊕ n)) :=
+    univ.filter (λ σ, ∀ x, ∃ y, sum.inl y = (σ (sum.inl x))),
```

Comment:
This set is `sum_congr_hom.range`, which might help golf the proof.

Reply:
In general, this proof shares a lot with the one in #5269; are you sure you can't make a more general statement about `(matrix.from_blocks A B C D).det`, and derive this special case from it?

Reply:
My experience is that proving the value of `det` is always rather involved - I certainly wouldn't advise to waste too much time to reduce the proof without setting up some more powerful proof methods first.